File: 1.Two_Sum/TwoSum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        map<int, int> m;
        for(int i = 0; i < nums.size(); ++i){
            int n = target - nums[i];
            if(m.count(n))
                return {m[n], i};
            m[nums[i]] = i;
        }
        
        return {-1, -1};
    }
};

File: 1.Two_Sum/TwoSum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        m = {}
        
        for i, num in enumerate(nums):
            temp = target - num
            
            if temp in m:
                return [m[temp], i]
            else:
                m[num] = i
